src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,104 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.AlreadyRegisteredException;
+import core.basesyntax.exceptions.ValidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final byte MIN_AUTHORIZATION_LENGTH = 6;
+    private static final byte MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws AlreadyRegisteredException, ValidDataException {
+        checkNulls(user);
+        isLoginNotRegistered(user);
+        isIdNotRegistered(user);
+        isLoginLengthValid(user);
+        isPasswordLengthValid(user);
+        isPasswordValid(user);
+        isLoginValid(user);
+        isAgeValid(user);
+        return storageDao.add(user);
+    }
+
+    private boolean checkNulls(User user) throws ValidDataException {
```

Comment:
your methods return a logical value, but you don't use it anywhere, you can replace it with void

Reply:
I missunderstanded that Bohdan say to use not void methods

Reply:
you are testing a method registration it is not void